pkg/universe.dagger.io/alpha/kubernetes/helm/upgrade.cue

```diff
@@ -125,11 +129,30 @@ import (
 					if dryRun {"--dry-run"},
 					if force {"--force"},
 					if cleanupOnFail {"--cleanup-on-fail"},
-					if username != _|_ {"--username=\(username)"},
-					if password != _|_ {"--password=${HELM_PASSWORD}"},
 				],
 				flags,
-			])
+			]),
+			" ")
+		env:  environment & {
+			HELM_CHART: chart
+			if password != _|_ {HELM_PASSWORD: password}
+			if username != _|_ {HELM_USERNAME: username}
 		}
+		entrypoint: ["sh", "-c"]
+		command: name: #"""
+            # if there is no password, run the command
+            if [ -z "$HELM_PASSWORD" ] || [ -z "$HELM_USERNAME" ]; then
+                \#(_cmd)
+                exit 0
+            fi
+            # otherwise check if we need to login to the registry or
+            # pass the credentials to the command
+            if [ "${HELM_CHART%%/*}" = "oci:" ]; then
```

Comment:
I also took care of logging into oci registries, since helm does not handle this part as expected.

Reply:
Nice